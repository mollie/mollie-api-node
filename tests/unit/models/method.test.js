import Method from 'models/method';

describe('method model', () => {
  it('should instantiate with default values', () => {
    const method = new Method();

    expect(method.getImage()).toBeNull();
    expect(method.toPlainObject()).toMatchSnapshot();
  });

  it('should instantiate with given values', () => {
    const methodProps = JSON.parse(`{
        "resource": "method",
        "id": "ideal",
        "description": "iDEAL",
        "image": {
            "size1x": "https://www.mollie.com/images/payscreen/methods/ideal.png",
            "size2x": "https://www.mollie.com/images/payscreen/methods/ideal%402x.png"
        },
        "_links": {
            "self": {
                "href": "https://api.mollie.com/v2/methods/ideal",
                "type": "application/hal+json"
            }
        }
    }`);
    const method = new Method(methodProps);

    expect(method.getImage('1x')).toBe(methodProps.image.size1x);
    expect(method.toPlainObject()).toMatchSnapshot();
  });
});
