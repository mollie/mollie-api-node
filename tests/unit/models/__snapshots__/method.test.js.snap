// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`method model should instantiate with default values 1`] = `
Object {
  "description": null,
  "id": null,
  "image": Object {
    "size1x": null,
    "size2x": null,
  },
  "resource": "method",
}
`;

exports[`method model should instantiate with given values 1`] = `
Object {
  "_links": Object {
    "self": Object {
      "href": "https://api.mollie.com/v2/methods/ideal",
      "type": "application/hal+json",
    },
  },
  "description": "iDEAL",
  "id": "ideal",
  "image": Object {
    "size1x": "https://www.mollie.com/images/payscreen/methods/ideal.png",
    "size2x": "https://www.mollie.com/images/payscreen/methods/ideal%402x.png",
  },
  "resource": "method",
}
`;
